Several fixes to rules and some Server terms

ggray-cb opened the pull request:
* Added some Server terms to the accepted words list.
* FIxed issues with the monospace and bold rules that would flag the region between two bolded or monospace passages in the same line. The oirignal regexp was too greedy and would match everything from the opening `` ` `` or `*` of the first passages to the closing `` ` `` or `*`. Now the matches end when they hit the closing character.
* The rule that flagged the word dialog being bolded had a similar issues where it would be triggered if the word "dialog" appeared between two instances of bolded text. For example: 'In the *Edit Bucket Settings* dialog, click *Show advanced bucket settings*.' The new rule only triggers if the opening `*` is preceded by a space, and the closing `*` is followed by a. space.

ggray-cb committed:
* Adds several Server terms to the accepted words list, including failover(s) and Hyperscale.
* FIxes issues with the Monospace and Bold rules where multiple instances of either in a single line would cause both instances and anything in betwen them to be highlighted. The new regexp is less greedy and stops matching one it hits the next ` or *.
* The rule that flags bolded instances of the word dialog was triggering if the word dialog appeared anywhere between two bolded pasages. I.e.: it would flag "In the *Edit Bucket Settings* dialog, click *Show advanced bucket settings*." even though "dialog" it outside the bold tags. The regexo now only triggers if the opening "*" is preceded by a whitespace and the closing "*" is followed by a whitespace. This ensures the two asterisks are part of the same bold tag, instead of being the end of one and the start of another.

ggray-cb committed:
For some reason, theses weren't adde to the previous push. See the commit message.

sarahlwelton commented on ValeStyles/config/vocabularies/Couchbase/accept.txt:
@ggray-cb do you want to account for capitalization on these terms? `accept.txt` is going to enforce lowercase matches only otherwise. Specifically only concerned about `failover` and `quantize`

ggray-cb replied:
OK, I updated `accept.txt` to add capitalization (and several forms of quantize).

sarahlwelton reviewed (commented):
Good tweaks on the `Bold`, `Dialogs`, and `Monospace` rules - I'm wondering if we should also limit the greediness of the capture for Italics. 

Just a question on your change to the `accept.txt` - don't want to cause more problems for you if you're starting a sentence with "Failover".

sarahlwelton commented:
I guess another thing - we had some Vale tests in place for what we wanted to flag with the "Dialogs" rule and now those seem to be failing. How'd it behave in your local testing @ggray-cb ?

ggray-cb commented:
@sarahlwelton I can;t seem to find instructions on running tests. I ran the test/runtests script, but that faled with an error with the article rules. I see more extensive tests in ValeStyles/test but no clear way to run them.

sarahlwelton commented:
@osfameron could you see about helping out with these tests when you have a chance? Are we supposed to be able to run these locally before making a PR?

osfameron commented:
Ok, so the PR check "Vale npm tests" is failing, with 2 errors on Couchbase.Dialogs and a few others.

For the first, you'll see it fails on:

* `Click *OK* in the *dialog* to continue.`
* `Capella displays the *dialog box*.`

So I suspect the regex isn't quite right. I'll have a play.

You should be able to run the tests locally with `npm test` from the ValeStyles/ directory.
(first `npm ci` to install the dependencies).

osfameron committed:
Test case and fix for Couchbase.Dialogs

`nonword` is required in order to be able to read the Asciidoc `*`.
Added test case, and tightened up regex to check we have nonspace inside the formatting.

osfameron committed:
skip span test (known failing issue with dasel)

osfameron commented:
I added a test case for Dialogs, confirmed that your new code handled that better than before, but broke the existing cases.
That's because the `nonword` is needed to parse the `*` markup.

The other failing test was my bad, fixed.
(I was trying to demonstrate a bug in Vale's upstream dependency `dasel` which causes bad highlighting in some odd cases.
A useful proof, but I should have then removed it to make our tests pass again... 🤦)

osfameron reviewed (approved):
Looks good with submitted correction 👍

osfameron committed:
Delete vestigial top-level test/

Superceded by ValeStyles/test/

osfameron commented:
Now documented at https://confluence.issues.couchbase.com/wiki/spaces/DOCS/pages/3463315517/Editing+and+Testing+Vale+styles#Running-the-tests%E2%80%A6

I've deleted the top-level `test/` directory in this PR, as it's superseded by `ValeStyles/test/` (and was confusing, as you noticed....)

ggray-cb commented:
> I've deleted the top-level `test/` directory in this PR, as it's superseded by `ValeStyles/test/` (and was confusing, as you noticed....)

Thanks! I'll address Sarah's other suggestion about capitalization.

ggray-cb committed:
Allowing Failover to be capitalized. Also added the npm package directory created by installing and running the test to gitignore.

ggray-cb committed:
Added a few more forms of quantize and allow it to be capitalized.

sarahlwelton reviewed (approved):
🚢
